analysis/topology/compact_open.lean

```diff
@@ -0,0 +1,149 @@
+/-
+Copyright (c) 2018 Reid Barton. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Reid Barton
+-/
+
+import analysis.topology.continuity tactic.tidy
+
+open set
+
+universes u v w
+
+section locally_compact
+
+-- There are various definitions of "locally compact space" in the
+-- literature, which agree for Hausdorff spaces but not in general.
+-- This one is the precise condition on X needed for the evaluation
+-- map C(X, Y) × X → Y to be continuous for all Y when C(X, Y) is
+-- given the compact-open topology.
+class locally_compact_space (α : Type u) [topological_space α] :=
+(local_compact_nhds : ∀ (x : α) (n ∈ (nhds x).sets), ∃ s ∈ (nhds x).sets, s ⊆ n ∧ compact s)
+
+variables {α : Type u} [topological_space α]
+
+lemma locally_compact_of_compact_nhds [t2_space α]
+  (h : ∀ x : α, ∃ s, s ∈ (nhds x).sets ∧ compact s) :
+  locally_compact_space α :=
+⟨assume x n hn,
+  let ⟨u, un, uo, xu⟩ := mem_nhds_sets_iff.mp hn in
+  let ⟨k, kx, kc⟩ := h x in
+  -- K is compact but not necessarily contained in N.
+  -- K \ U is again compact and doesn't contain x, so
+  -- we may find open sets V, W separating x from K \ U.
+  -- Then K \ W is a compact neighborhood of x contained in U.
```

Comment:
I'm very interested in knowing whether such comments are welcome in mathlib. I think it's a good idea, but I never dared.

Reply:
Please can we have such comments welcome? (Perhaps even encouraged in the style guide?)

Reply:
I think it is good to have these kind of comments, but I a problem is that the comment doesn't fit the formalization (in this case it is just `K`->`k` etc. ). But this can easily deteriorate with time (other name changes, generalizations etc...).
Wouldn't this comment make more sense at `compact_compact_separated`?

Another problem is that `let` usually doesn't allow us to annotate types. Then this would be more readable, as we would at least see what we derive.